code/game/objects/items/toys.dm

```diff
@@ -904,8 +909,248 @@
+	icon_state = "plushie_borg"
+	var/on = FALSE
+
+/obj/item/toy/plushie/borgplushie/Initialize(mapload)
+	. = ..()
+	add_overlay(pick("plushie_borgjan", "plushie_borgsec", "plushie_borgmed", "plushie_borgmine", "plushie_borgserv", "plushie_borgassist", "plushie_borgengi"))
+
+/obj/item/toy/plushie/borgplushie/attack_self(mob/user)
+	on = !on
+	update_brightness()
+	return ..()
+
+/obj/item/toy/plushie/borgplushie/proc/update_brightness()
+	if(on)
+		set_light(4)
+	else
+		set_light(0)
+	update_icon()
+
+/obj/item/toy/plushie/borgplushie/update_overlays()
+	. = ..()
+	if(on)
+		add_overlay("borglights")
+	else
+		cut_overlay("borglights")
+
+/obj/item/toy/plushie/borgplushie/extinguish_light(force = FALSE)
+	if(!force)
+		if(on)
+			visible_message("<span class='danger'>[src]'s light grows dim...</span>")
+			on = !on
+			update_brightness()
+	else
+		atom_say("Self-destruct command received!</span>")
+		visible_message("<span class='danger'>[src] explodes!</span>")
+		var/turf/T = get_turf(src)
+		playsound(T, 'sound/goonstation/effects/robogib.ogg', 50, 1)
+		robogibs(T)
+		if(grenade)
+			explosive_betrayal(grenade)
+		if(!QDELETED(src))
+			qdel(src)
+
+/obj/item/toy/plushie/dionaplushie
+	name = "diona plushie"
+	desc = "This plushy is seemingly comprised of other, smaller, nymph plushies. They really went all out on the realism! Keep away from plantkiller."
+	icon_state = "plushie_diona"
+	COOLDOWN_DECLARE(creak_cooldown)
+
+/obj/item/toy/plushie/dionaplushie/attack_self(mob/user)
+	if(!COOLDOWN_FINISHED(src, creak_cooldown))
+		return ..()
+	playsound(src,'sound/voice/dionatalk1.ogg', 10, 0)
+	visible_message("<span class='danger'>Creak...</span>")
+	COOLDOWN_START(src, creak_cooldown, 3 SECONDS)
+	return
+
+/obj/item/toy/plushie/nymphplushie
+	name = "nymph plushie"
+	desc = "Life-sized plushie of a diona nymph, perhaps if you find another you could make a diona!"
+	icon_state = "plushie_nymph"
+	COOLDOWN_DECLARE(chirp_cooldown)
+
+/obj/item/toy/plushie/nymphplushie/attackby(obj/item/B, mob/user, params)
+	if(istype(B, /obj/item/toy/plushie/nymphplushie))
+		var/obj/item/toy/plushie/nymphplushie/NP = B
+		var/found_grenade = FALSE
+		if(grenade)
+			found_grenade = TRUE
+			explosive_betrayal(grenade)
+		if(NP.grenade)
+			found_grenade = TRUE
+			NP.explosive_betrayal(NP.grenade)
+		if(found_grenade)
+			return
+		new /obj/item/toy/plushie/dionaplushie(get_turf(loc))
+		to_chat(user, "<span class='notice'>The nymph plushies combine seamlessly into an diona plushie!</span>")
+		playsound(loc, 'sound/voice/dionatalk1.ogg', 50, 1)
+		qdel(NP)
+		qdel(src)
+	else
+		return ..()
+
+/obj/item/toy/plushie/nymphplushie/attack_self(mob/user)
+	if(!COOLDOWN_FINISHED(src, chirp_cooldown))
+		return ..()
+	playsound(src,'sound/creatures/nymphchirp.ogg', 10, 0)
+	visible_message("<span class='danger'>Chirp!</span>")
+	COOLDOWN_START(src, chirp_cooldown, 3 SECONDS)
+	return
```

Comment:
This is a lot of code repetition. You need to add this to the plushie basetype, with two vars.
```dm
var/sound/rare_hug_sound
var/rare_hug_word
COOLDOWN_DECLARE(rare_hug_cooldown)
```
and then set them on the subtypes like
```dm
rare_hug_sound = 'sound/creatures/nymphchirp.ogg'
rare_hug_word = "Chirp!"
```
and on the base code have
```dm
/obj/item/toy/plushie/attack_self(mob/user as mob)
	if(rare_hug_sound && rare_hug_word && COOLDOWN_FINISHED(src, rare_hug_cooldown))
		playsound(src, rare_hug_sound , 10, 0)
		visible_message("<span class='danger'>[rare_hug_word]</span>")
		COOLDOWN_START(src, rare_hug_cooldown, 3 SECONDS)
		return ..()
	
	if(has_stuffing || grenade)
		var/cuddle_verb = pick("hugs", "cuddles", "snugs")
		user.visible_message("<span class='notice'>[user] [cuddle_verb] [src].</span>")
		playsound(get_turf(src), poof_sound, 50, TRUE, -1)
		if(grenade && !grenade.active)
			add_attack_logs(user, user, "activated a hidden grenade in [src].", ATKLOG_MOST)
			playsound(loc, 'sound/weapons/armbomb.ogg', 10, TRUE, -3)
			//We call with grenade as argument, so cutting the grenade out doesn't magically defuse it
			addtimer(CALLBACK(src, PROC_REF(explosive_betrayal), grenade), rand(1, 3) SECONDS)
	else
		to_chat(user, "<span class='notice'>You try to pet [src], but it has no stuffing. Aww...</span>")
	return ..()

Reply:
Fixed